File: dvbViewerController/src/main/java/org/dvbviewer/controller/io/data/FFMPEGPrefsHandler.java
package org.dvbviewer.controller.io.data;

import org.apache.commons.lang3.StringUtils;
import org.dvbviewer.controller.entities.FfMpegPrefs;
import org.dvbviewer.controller.entities.Preset;
import org.dvbviewer.controller.ui.fragments.StreamConfig;
import org.dvbviewer.controller.utils.INIParser;

import java.util.Iterator;

/**
 * Created by karlw on 18.12.2015.
 */
public class FFMPEGPrefsHandler {

    public FfMpegPrefs parse(String ffmpegprefs) throws Exception {
        FfMpegPrefs ffPrefs = new FfMpegPrefs();
        INIParser iniParser = new INIParser(ffmpegprefs);
        ffPrefs.setVersion(iniParser.getString("Version", "Version"));
        Iterator<String> sectionIterator = iniParser.getSections();
        while (sectionIterator.hasNext()) {
            String sectionName = sectionIterator.next();
            if (isPreset(iniParser, sectionName)) {
                Preset preset = new Preset();
                preset.setTitle(sectionName);
                final String mimeType = iniParser.getString(sectionName, "MimeType");
                if (StringUtils.isEmpty(mimeType)) {
                    preset.setMimeType(StreamConfig.M3U8_MIME_TYPE);
                } else {
                    preset.setMimeType(mimeType);
                }
                preset.setExtension(iniParser.getString(sectionName, "Ext"));
                ffPrefs.getPresets().add(preset);
            }
        }
        return ffPrefs;
    }

    private boolean isPreset(INIParser iniParser, String sectionName) {
        Iterator<String> keysIterator = iniParser.getKeys(sectionName);
        boolean isPreset = false;
        while(keysIterator.hasNext()){
            String keyName = keysIterator.next();
            if ("Cmd".equals(keyName)){
                isPreset = true;
            }
        }
        return isPreset;
    }
}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/api/handler/FFMPEGPrefsHandler.kt
package org.dvbviewer.controller.data.api.handler


import org.apache.commons.lang3.StringUtils
import org.dvbviewer.controller.data.entities.FFMpegPresetList
import org.dvbviewer.controller.data.entities.Preset
import org.dvbviewer.controller.utils.INIParser
import org.dvbviewer.controller.utils.StreamUtils

class FFMPEGPrefsHandler {

    @Throws(Exception::class)
    fun parse(ffmpegprefs: String?): FFMpegPresetList {
        val ffPrefs = FFMpegPresetList()
        if (StringUtils.isBlank(ffmpegprefs)) {
            return ffPrefs
        }
        val iniParser = INIParser(ffmpegprefs)
        ffPrefs.version = iniParser.getString("Version", "Version")
        val sectionIterator = iniParser.sections
        while (sectionIterator.hasNext()) {
            val sectionName = sectionIterator.next()
            if (isPreset(iniParser, sectionName)) {
                val preset = Preset()
                preset.title = sectionName
                val mimeType = iniParser.getString(sectionName, "MimeType")
                if (StringUtils.isEmpty(mimeType)) {
                    preset.mimeType = StreamUtils.M3U8_MIME_TYPE
                } else {
                    preset.mimeType = mimeType
                }
                preset.extension = iniParser.getString(sectionName, "Ext")
                ffPrefs.presets.add(preset)
            }
        }
        return ffPrefs
    }

    private fun isPreset(iniParser: INIParser, sectionName: String): Boolean {
        val keysIterator = iniParser.getKeys(sectionName)
        var isPreset = false
        while (keysIterator!!.hasNext()) {
            val keyName = keysIterator.next()
            if ("Cmd" == keyName) {
                isPreset = true
            }
        }
        return isPreset
    }
}
